docs(Vue Testing Library): update `@vue/test-utils` link urls in api page to vue3

ytak-sagit opened the pull request:
# Change Summary

This pull request updates several links on the Vue Testing Library API page.

Previously, these links for `@vue/test-utils` were directing users to the documentation site for Vue v2.x and earlier.  
Since today's Vue version is 3.x, I update these links to the `@vue/test-utils` documentation site for Vue 3.

timdeschryver reviewed (commented):
Looks good, thanks @ytak-sagit !
There seems to be 2 more references to https://vue-test-utils on the FAQ page, do you also want to update these?

ytak-sagit commented:
@timdeschryver  
Thanks for check and comments!  
I had missed the FAQ page, so I updated its links as well at 7c31536 .

Also, I updated the following two points:
- [5537ca5] Updated the branch name in the stub-example links from `master` to `main` to prevent redirects.
- [5de7aff] Changed the key name to `global.stubs` to match the `@vue/test-utils` documentation site for Vue 3.
